tasks/service/async/src/main/java/org/projectnessie/nessie/tasks/async/vertx/VertxTasksAsync.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (C) 2024 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.nessie.tasks.async.vertx;
+
+import io.vertx.core.Vertx;
+import java.time.Clock;
+import java.time.Instant;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.CompletionStage;
+import java.util.function.Supplier;
+import org.projectnessie.nessie.tasks.async.ScheduledHandle;
+import org.projectnessie.nessie.tasks.async.TasksAsync;
+
+public class VertxTasksAsync implements TasksAsync {
+  private final Vertx vertx;
+  private final Clock clock;
+  private final long minimumDelayMillis;
+
+  public VertxTasksAsync(Vertx vertx, Clock clock, long minimumDelayMillis) {
+    this.vertx = vertx;
+    this.clock = clock;
+    this.minimumDelayMillis = minimumDelayMillis;
+  }
+
+  @Override
+  public CompletionStage<Void> call(Runnable runnable) {
```

Comment:
This impl could be promoted to default impl at interface level.

Reply:
Why? those are different for every backend.

Reply:
Because it seems like the natural way to implement `call` as a special case of `supply`.